### PR TITLE
Add an operator e2e exclusion tests when a single log process has high latency

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Add a new e2e test to see how exclusions behave if a log process in the remote region has high latencies.

## Type of change

- Other

## Discussion

-

## Testing

CI will run the test and I'm running it manually.

## Documentation

Will be added to the test case.

## Follow-up

-

### CODE REVIEW

File: e2e/test_operator_ha/operator_ha_test.go [hunk 1]
```diff
@@ -211,4 +213,109 @@ var _ = Describe("Operator HA tests", Label("e2e", "pr"), func() {
...
+		})
+
+		When("when a remote log has network latency issues and gets replaced", func() {
+			var experiment *fixtures.ChaosMeshExperiment
+
+			BeforeEach(func() {
+				spec := fdbCluster.GetRemote().GetCluster().Spec.DeepCopy()
+				spec.AutomationOptions.UseLocalitiesForExclusion = pointer.Bool(true)
+				fdbCluster.GetRemote().UpdateClusterSpecWithSpec(spec)
+				Expect(fdbCluster.GetRemote().GetCluster().UseLocalitiesForExclusion()).To(BeTrue())
```
Comment (nicmorales9) [inline e2e/test_operator_ha/operator_ha_test.go:241]: seems duplicated from the beforeEach above?

**Review by nicmorales9 — APPROVED** [review]

File: e2e/test_operator_ha/operator_ha_test.go (hunk 1 above)
Comment (johscheuer) [reply e2e/test_operator_ha/operator_ha_test.go:241]: Good catch, I copied that around and missed that one 👍

**Review by spraza — APPROVED** [review]

File: e2e/test_operator_ha/operator_ha_test.go [hunk 2]
```diff
@@ -211,4 +213,105 @@ var _ = Describe("Operator HA tests", Label("e2e", "pr"), func() {
...
+					replacedPod = pod
+					break
+				}
+
+				log.Println("Inject latency chaos")
+				experiment = factory.InjectNetworkLatency(
+					fixtures.PodSelector(&replacedPod),
+					chaosmesh.PodSelectorSpec{
+						GenericSelectorSpec: chaosmesh.GenericSelectorSpec{
+							Namespaces:     []string{fdbCluster.GetRemote().Namespace()},
```
Comment (johscheuer) [inline e2e/test_operator_ha/operator_ha_test.go:281]: Discussed offline, we also want to have a test where the primary and remote side have latencies injected.

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [d735e6b](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/d735e6baa503caa8b8b7c74475ac39e67a5de200)
> * Duration 3:21:07
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/180b552d-8fca-4701-b675-42e965b79872.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/180b552d-8fca-4701-b675-42e965b79872/workspace.zip) zip file of the working directory (available for 30 days)

The test failure was unrelated and was an issue with our cache.

**Pull request merged**